conan/tools/cmake/layout.py

```diff
@@ -63,3 +65,49 @@ def get_build_folder_custom_vars(conanfile):
 
     user_defined_build = "settings.build_type" in build_vars
     return "-".join(ret), user_defined_build
+
+
+def get_root_test_build_folder(conanfile):
+    """
+    Gets a different root test build folder path (or the test_package/ one) if
+    tools.cmake.cmake_layout:test_build_folder is specified.
+
+    If specified the test output build folder could look like this:
+
+    >> [TEST_ROOT_PATH]/test_output-[BUILD_TYPE]_[COMPILER]_[COMPILER_VERSION]-[SHA1]
+
+    Several notes:
+
+        * tools.cmake.cmake_layout:test_build_folder=recipe_folder indicates that the root path
+          will be the test_package/ by default (as usual).
+        * tools.cmake.cmake_layout:test_build_folder=MY_FOLDER/ will add to the test build folder
+          name the [REF_NAME]_[REF_VERSION] as prefix, so it could look like:
+          >> [TEST_ROOT_PATH]/[REF_NAME]_[REF_VERSION]-test_output-[BUILD_TYPE]_[COMPILER]_[COMPILER_VERSION]-[SHA1]
+        * If tools.cmake.cmake_layout:test_build_folder is not specified, it will work as usual.
+    """
+    # Get root build folder from global.conf or profile
+    test_build_root_folder = conanfile.conf.get("tools.cmake.cmake_layout:test_build_folder",
+                                                check_type=str)
+    tested_reference = conanfile.tested_reference_str
+    # FIXME: Should we have a better way to check that?
+    if tested_reference is None or test_build_root_folder is None:
```

Comment:
I am afraid this is not the expected behavior that we were discussing in the issue. The strategy should always return a different folder than it is doing right now, which is broken. Not only when ``tools.cmake.cmake_layout:test_build_folder`` is defined, but always, and ``tools.cmake.cmake_layout:test_build_folder`` is used to be able to use a different location, like ``C:/mytemp`` for the test_package build directories. But this PR should be creating always the layout:
```
conanfile.py
test_package
   | - conanfile.py
   | - build
          | - gcc10x64debug_af2
          | - msvc191x86release_b21
```
Every folder gets deleted just before being re-created
Then, if ``tools.cmake.cmake_layout:test_build_folder=C:/mytemp`` the layout would be something like:
```
conanfile.py
test_package
   | - conanfile.py

C:/mytemp
   | - mypkg/myversion/recipe_revision_firstNchars/test
                                                       | - build
                                                              | - gcc10x64debug_af2
                                                              | - msvc191x86release_b21
```